libtenzir/builtins/aggregation-functions/mode_value_counts_entropy.cpp

```diff
@@ -46,29 +45,48 @@ class instance final : public aggregation_instance {
   }
 
   auto get() const -> data override {
-    if constexpr (Kind == kind::mode) {
-      const auto comp = [](const auto& lhs, const auto& rhs) {
-        return lhs.second < rhs.second;
-      };
-      const auto it = std::ranges::max_element(counts_, comp);
-      if (it == counts_.end()) {
-        return {};
+    switch (Kind) {
+      case kind::mode: {
+        const auto comp = [](const auto& lhs, const auto& rhs) {
+          return lhs.second < rhs.second;
+        };
+        const auto it = std::ranges::max_element(counts_, comp);
+        if (it == counts_.end()) {
+          return {};
+        }
+        return it->first;
       }
-      return it->first;
-    } else {
-      auto result = list{};
-      result.reserve(counts_.size());
-      for (const auto& [value, count] : counts_) {
-        result.emplace_back(record{
-          {"value", value},
-          {"count", count},
+      case kind::value_counts: {
+        auto result = list{};
+        result.reserve(counts_.size());
+        for (const auto& [value, count] : counts_) {
+          result.emplace_back(record{
+            {"value", value},
+            {"count", count},
+          });
+        }
+        std::ranges::sort(result, std::less<>{}, [](const auto& x) {
+          return as_vector(as<record>(x))[0].second;
         });
+        return result;
+      }
+      case kind::entropy: {
+        auto result = 0.0;
+        const auto total
+          = std::transform_reduce(counts_.begin(), counts_.end(), size_t{},
+                                  std::plus<>{}, [](const auto& x) {
+                                    return x.second;
+                                  });
+        for (const auto& [_, count] : counts_) {
+          const auto p_x = detail::narrow<double>(count) / total;
+          if (p_x > 0.0) {
+            result -= p_x * std::log2(p_x);
+          }
+        }
+        return result;
```

Comment:
Note for the reviewer: This is the actual implementation, the rest is mostly refactoring.